vsintegration/src/FSharp.Editor/CodeFixes/RemoveUnnecessaryParentheses.fs

```diff
@@ -137,11 +174,38 @@ type internal FSharpRemoveUnnecessaryParenthesesCodeFixProvider [<ImportingConst
                                     match line.AsSpan(startCol).IndexOfAnyExcept(' ', ')') with
                                     | -1 -> loop innerOffsides (lineNo + 1) 0
                                     | i ->
-                                        match innerOffsides with
-                                        | NoneYet -> loop (FirstLine(i + startCol)) (lineNo + 1) 0
-                                        | FirstLine innerOffsides -> loop (FollowingLine(innerOffsides, i + startCol)) (lineNo + 1) 0
-                                        | FollowingLine(firstLine, innerOffsides) ->
-                                            loop (FollowingLine(firstLine, min innerOffsides (i + startCol))) (lineNo + 1) 0
+                                        match line[i + startCol ..] with
+                                        | StartsWithMatch
+                                        | StartsWithSingleLineComment -> loop innerOffsides (lineNo + 1) 0
+                                        | StartsWithMultilineComment depth ->
+                                            let rec skip depth lineNo =
+                                                if lineNo <= endLineNo then
+                                                    let line = sourceText.Lines[lineNo].ToString()
+
+                                                    let delta =
+                                                        line.TrimStart ' '
+                                                        |> Seq.pairwise
+                                                        |> Seq.sumBy (function
+                                                            | '(', '*' -> 1
+                                                            | '*', ')' -> -1
+                                                            | _ -> 0)
+
+                                                    if depth + delta = 0 then
+                                                        loop innerOffsides (lineNo + 1) 0
+                                                    else
+                                                        skip depth (lineNo + 1)
+                                                else
+                                                    innerOffsides
+
+                                            skip depth (lineNo + 1)
```

Comment:
Looks like I accidentally included this... I should probably just take it out for now, even though it would work for some scenarios.

Really, we'd need something fully aware of comments, strings, ifdefs, etc...

Alternatively, we could just be more conservative in the logic for `containsSensitiveIndentation` in SynExpr.fs, and always err on the side of caution if we encounter comments, strings, ifdefs, etc.

Reply:
c16e33d